File: listeners/who_asked.js
module.exports = {
    event: 'message',
    callback: async (message) => {
        if (/who ([^ ]+ ){0,2}asked/gi.test(message.content)) {
            let msg = await message.channel.send('https://youtu.be/nFrPQ86kxo4');
            msg.suppressEmbeds(true);
        }
    }
}

File: lib/who_asked.dart
import 'package:nyxx/nyxx.dart';

final pattern = RegExp(r'who (\w+ ){0,2}asked');

class WhoAsked extends NyxxPlugin<NyxxGateway> {
  @override
  String get name => 'WhoAsked';

  @override
  void afterConnect(NyxxGateway client) {
    client.onMessageCreate.listen((event) async {
      if (pattern.hasMatch(event.message.content)) {
        await event.message.channel.sendMessage(MessageBuilder(
          content: 'https://www.youtube.com/watch?v=83amztBl9Qk&t=211s',
        ));
      }
    });
  }
}
